api/config/v1alpha1/shared_types.go

```diff
@@ -67,6 +82,14 @@ type KubernetesServiceSpec struct {
 	// +optional
 	Annotations map[string]string `json:"annotations,omitempty"`
 
+	// Type determines how the Service is exposed. Defaults to LoadBalancer.
+	// Valid options are ClusterIP and LoadBalancer.
+	// "LoadBalancer" means a service will be exposed via an external load balancer (if the cloud provider supports it).
+	// "ClusterIP" means a service will only be accessible inside the cluster, via the cluster IP.
+	// +kubebuilder:default:="LoadBalancer"
+	// +optional
+	Type *ServiceType `json:"type,omitempty"`
```

Comment:
if `ServiceType` is optional, default value will not work, after following CR applied
```console
apiVersion: config.gateway.envoyproxy.io/v1alpha1
kind: EnvoyProxy
metadata:
  name: example-config
spec:
  provider:
    type: Kubernetes
    kubernetes:
      envoyService: {}
```

you can check EnvoyProxy by running:
```
k get envoyproxy -oyaml example-config 
apiVersion: config.gateway.envoyproxy.io/v1alpha1
kind: EnvoyProxy
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"config.gateway.envoyproxy.io/v1alpha1","kind":"EnvoyProxy","metadata":{"annotations":{},"name":"example-config","namespace":"default"},"spec":{"provider":{"kubernetes":{"envoyService":{}},"type":"Kubernetes"}}}
  creationTimestamp: "2023-03-24T03:09:19Z"
  generation: 1
  name: example-config
  namespace: default
  resourceVersion: "236531"
  uid: 81deaa89-0f6f-41b4-a17c-dba8a70ae834
spec:
  logging:
    level:
      system: info
  provider:
    kubernetes:
      envoyService:
        type: LoadBalancer
    type: Kubernetes
```

Reply:
the default value is `LoadBalancer ` and after following CR applied the type same as the default. 

what's the problem?

Reply:
my mistake, tested with a wrong commit.